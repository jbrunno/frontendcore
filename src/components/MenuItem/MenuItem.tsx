import React from 'react';
import {
  MenuItem as MuiMenuItem,
  MenuItemProps as MuiMenuItemProps,
} from '@mui/material';

export type MenuItemProps = MuiMenuItemProps;

export function MenuItem(props: MenuItemProps) {
  return <MuiMenuItem {...props} />;
}
export default MenuItem;
