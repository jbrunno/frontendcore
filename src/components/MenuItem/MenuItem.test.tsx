import React from 'react';
import { render } from '@test-utils/component-testing';
import { MenuItem } from './MenuItem';

test('renders the MenuItem', () => {
  const { container } = render(<MenuItem>Item</MenuItem>);
  expect(container).toMatchSnapshot();
});
