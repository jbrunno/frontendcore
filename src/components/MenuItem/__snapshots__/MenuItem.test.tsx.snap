// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MenuItem 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0000000A;
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-selected {
  background-color: rgba(74, 49, 255, 0.08);
}

.emotion-0.Mui-selected.Mui-focusVisible {
  background-color: rgba(74, 49, 255, 0.2);
}

.emotion-0.Mui-selected:hover {
  background-color: rgba(74, 49, 255, 0.12);
}

@media (hover: none) {
  .emotion-0.Mui-selected:hover {
    background-color: rgba(74, 49, 255, 0.08);
  }
}

.emotion-0.Mui-focusVisible {
  background-color: #0000001F;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
}

.emotion-0+.MuiDivider-root {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.emotion-0+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-0 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-0 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-0 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: auto;
  }
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <li
    class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters emotion-0"
    role="menuitem"
    tabindex="-1"
  >
    Item
    <span
      class="MuiTouchRipple-root emotion-1"
    />
  </li>
</div>
`;
